shell/platform/windows/platform_handler_win32_unittests.cc

```diff
@@ -0,0 +1,239 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/platform_handler_win32.h"
+
+#include <memory>
+
+#include "flutter/shell/platform/common/json_method_codec.h"
+#include "flutter/shell/platform/windows/testing/mock_window_binding_handler.h"
+#include "flutter/shell/platform/windows/testing/test_binary_messenger.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "rapidjson/document.h"
+
+namespace flutter {
+namespace testing {
+
+namespace {
+using ::testing::_;
+
+static constexpr char kChannelName[] = "flutter/platform";
+
+static constexpr char kHasStringsClipboardMethod[] = "Clipboard.hasStrings";
+
+static constexpr char kTextPlainFormat[] = "text/plain";
+
+static constexpr char kValueKey[] = "value";
+static constexpr int kAccessDeniedErrorCode = 5;
+static constexpr int kErrorSuccess = 0;
+
+}  // namespace
+
+// A test version of the private ScopedClipboard.
+class TestScopedClipboard : public ScopedClipboardInterface {
+ public:
+  TestScopedClipboard(int open_error, bool has_strings);
```

Comment:
Nit: bool elements are really hard to read at the call site for non-trivial cases (e.g., `SetSomeBool(bool)`); consider making an enum instead (`enum class ClipboardState { kHasString, kNoString }`) and using that for the argument type.